File: MATTHEW/leetcode/AIcoding/4.GPT-4o/go_solutions/3364-minimum-positive-sum-subarray.go
func minPositiveSumSubarray(nums []int) int {
    minSum := int(^uint(0) >> 1)
    currentSum := 0
    foundPositive := false

    for _, num := range nums {
        if num > 0 {
            foundPositive = true
            currentSum += num
            if currentSum < minSum {
                minSum = currentSum
            }
        } else {
            currentSum = 0
        }
    }

    if !foundPositive {
        return 0
    }
    return minSum
}

File: leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/3364-minimum-positive-sum-subarray.js
var minSubArray = function(nums) {
    let minSum = Infinity;
    let currentSum = 0;
    for (let num of nums) {
        currentSum += num;
        if (currentSum > 0) {
            currentSum = 0;
        } else if (currentSum < minSum) {
            minSum = currentSum;
        }
    }
    return minSum === Infinity ? 0 : minSum;
};